src/worker.cpp

```diff
@@ -1335,4 +1335,61 @@ void Worker::performBackupAndRestore(types::VPDMapVariant& io_srcVpdMap)
         */
     }
 }
+
+void Worker::deleteFruVpd(const std::string& i_dbusObjPath)
+{
+    try
+    {
+        bool l_presentPropValue = std::get<bool>(dbusUtility::readDbusProperty(
+            constants::pimServiceName, i_dbusObjPath,
+            constants::inventoryItemInf, "Present"));
+
+        if (!l_presentPropValue)
+        {
+            logging::logMessage(
+                "Given FRU is not present. Could not perform deletion for " +
+                i_dbusObjPath);
+            return;
+        }
+        else
+        {
+            if (jsonUtility::isActionRequired(m_parsedJson,
+                                              jsonUtility::getFruPathFromJson(
+                                                  m_parsedJson, i_dbusObjPath),
+                                              "preAction", "deletion"))
+            {
+                if (!jsonUtility::processSystemCmdTag(
+                        m_parsedJson,
+                        jsonUtility::getFruPathFromJson(m_parsedJson,
+                                                        i_dbusObjPath),
+                        "preAction", "deletion"))
+                {
+                    throw std::runtime_error("Failed to unbind the LED driver");
+                }
+            }
+
+            types::InterfaceMap l_interfaceMap;
+            vpdSpecificUtility::clearVpdOnDbus(i_dbusObjPath,
+                                                  l_interfaceMap);
+
+            types::ObjectMap l_objectMap;
+            l_objectMap.emplace(i_dbusObjPath, std::move(l_interfaceMap));
+
+            if (!dbusUtility::callPIM(std::move(l_objectMap)))
+            {
+                throw std::runtime_error("call to PIM failed for: " +
+                                         i_dbusObjPath);
+            }
+        }
+
+        logging::logMessage("Successfully completed deletion of VPD for " +
+                            i_dbusObjPath);
+    }
+    catch (const std::exception& l_ex)
+    {
+        // ToDo - log PEL
+        logging::logMessage("Failed to delete FRU VPD with error: " +
```

Comment:
can we add for which fru delete failed ?

Reply:
Done